experiment/simulation/js/factorial.js

```diff
@@ -0,0 +1,335 @@
+    basecaseOperatorSelect.addEventListener('change', function () {
+        basecaseOperator = changeOperator(basecaseOperatorSelect.value);
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`basecase-operator-${language1}`).value = basecaseOperatorSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let basecaseSelect = document.getElementById(`basecase-${language}`);
+    basecaseSelect.addEventListener('change', function () {
+        basecase = parseInt(basecaseSelect.value);
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`basecase-${language1}`).value = basecaseSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let basecaseReturnSelect = document.getElementById(`basecase-return-${language}`);
+    basecaseReturnSelect.addEventListener('change', function () {
+        basecaseReturn = parseInt(basecaseReturnSelect.value);
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`basecase-return-${language1}`).value = basecaseReturnSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let recursecaseSelect = document.getElementById(`recursecase-${language}`);
+    recursecaseSelect.addEventListener('change', function () {
+        recursecase = recursecaseSelect.value;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`recursecase-${language1}`).value = recursecaseSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let variableSelect = document.getElementById(`variable-${language}`);
+    variableSelect.addEventListener('change', function () {
+        variable = variableSelect.value;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`variable-${language1}`).value = variableSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let operatorSelect = document.getElementById(`operator-${language}`);
+    operatorSelect.addEventListener('change', function () {
+        operator = operatorSelect.value;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`operator-${language1}`).value = operatorSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let nSelect = document.getElementById(`n-${language}`);
+    nSelect.addEventListener('change', function () {
+        maxStackDepth = parseInt(nSelect.value);
+        stackText.textContent = maxStackDepth;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`n-${language1}`).value = nSelect.value;
+        })
+    })
+});
+
+function step() {
+    executedsteps++;
+    if (stepstoexecute != -1 && executedsteps > stepstoexecute)
+        return false;
+    return true;
+}
+
+function changeOperator(operator) {
```

Comment:
This function is common across 2 files, can you see if you can create a common utility file and define all such functions there?